### PR TITLE
COMMERCE-12309

### PR DESCRIPTION
**Pull request by alessiorendina**
https://liferay.atlassian.net/browse/COMMERCE-12309

Based on #4323

### CODE REVIEW

**Comment by alessiorendina** [comment]
ci:test:sf

**Comment by alessiorendina** [comment]
ci:test:sf

**Comment by alessiorendina** [comment]
ci:test:sf

**Comment by alessiorendina** [comment]
ci:test:sf

**Comment by alessiorendina** [comment]
ci:test:sf

**Comment by alessiorendina** [comment]
@andrea-ale-sbarra this is now ready for review

**Comment by alessiorendina** [comment]
ci:test:sf

**Comment by alessiorendina** [comment]
ci:test:sf

File: modules/apps/commerce/commerce-frontend-js/src/main/java/com/liferay/commerce/frontend/js/internal/servlet/taglib/CommerceFrontendJsDynamicInclude.java [hunk 1]
```diff
@@ -163,6 +164,18 @@ private String _getContent(
 					return commerceOrderCheckoutConfiguration.
 						showSeparateOrderItems();
 				}
+			).put(
```
Comment (andrea-ale-sbarra) [inline modules/apps/commerce/commerce-frontend-js/src/main/java/com/liferay/commerce/frontend/js/internal/servlet/taglib/CommerceFrontendJsDynamicInclude.java:167]: I'm not sure about this, this way this configuration is only accessible via our code, it's not something accessible via an API

**Review by andrea-ale-sbarra — COMMENTED** [review]
Overall it looks good.
I'm not sure if we really want to limit this feature to who uses our js, I think this config can be added to the product entity of the delivery api

File: modules/apps/commerce/commerce-frontend-js/src/main/java/com/liferay/commerce/frontend/js/internal/servlet/taglib/CommerceFrontendJsDynamicInclude.java (hunk 1 above)
Comment (andrea-ale-sbarra) [reply modules/apps/commerce/commerce-frontend-js/src/main/java/com/liferay/commerce/frontend/js/internal/servlet/taglib/CommerceFrontendJsDynamicInclude.java:167]: That was my concern indeed; I was questioning if it is right to have that configuration at instance level, but it's a requirement so it's right.

**Review by andrea-ale-sbarra — APPROVED** [review]
LGTM.
Thank you!

**Comment by andrea-ale-sbarra** [comment]
ci:test:relevant

**Comment by andrea-ale-sbarra** [comment]
ci:test:commerce-functional

**Comment by andrea-ale-sbarra** [comment]
ci:test:relevant

**Comment by alessiorendina** [comment]
ci:test:relevant

**Comment by alessiorendina** [comment]
ci:test:sf

**Comment by alessiorendina** [comment]
ci:test:relevant

**Comment by alessiorendina** [comment]
ci:test:sf

**Comment by alessiorendina** [comment]
ci:test:sf

**Comment by alessiorendina** [comment]
ci:test:sf

**Comment by alessiorendina** [comment]
ci:test:sf

**Comment by alessiorendina** [comment]
ci:test:sf

**Comment by alessiorendina** [comment]
ci:test:sf

File: modules/apps/commerce/commerce-product-service/src/main/java/com/liferay/commerce/product/internal/validator/ProductsLimitCommerceOrderValidatorImpl.java [hunk 2]
```diff
@@ -0,0 +1,229 @@
...
+
+	@Override
+	public String getKey() {
+		return KEY;
+	}
+
+	@Override
+	public CommerceOrderValidatorResult validate(
+			Locale locale, CommerceOrder commerceOrder, CPInstance cpInstance,
+			String json, BigDecimal quantity)
```
Comment (andrea-ale-sbarra) [inline modules/apps/commerce/commerce-product-service/src/main/java/com/liferay/commerce/product/internal/validator/ProductsLimitCommerceOrderValidatorImpl.java:66]: child?

**Review by andrea-ale-sbarra — CHANGES_REQUESTED** [review]
Just 2 checks.
Overall looks good.
Thank you!

**Comment by alessiorendina** [comment]
ci:test:sf

**Comment by alessiorendina** [comment]
ci:test:relevant

**Comment by alessiorendina** [comment]
ci:test:sf

**Comment by alessiorendina** [comment]
ci:test:relevant

**Comment by alessiorendina** [comment]
ci:test:sf

**Comment by alessiorendina** [comment]
ci:test:relevant

**Comment by alessiorendina** [comment]
ci:test:sf

**Comment by alessiorendina** [comment]
ci:test:relevant

**Comment by alessiorendina** [comment]
ci:test:sf

**Comment by alessiorendina** [comment]
ci:test:relevant

**Comment by alessiorendina** [comment]
ci:test:sf

**Comment by alessiorendina** [comment]
ci:test:sf

**Comment by alessiorendina** [comment]
ci:test:relevant

**Comment by 22LucaGiannotta22** [comment]
ci:test:commerce-functional

**Comment by 22LucaGiannotta22** [comment]
ci:test:commerce-functional

**Comment by alessiorendina** [comment]
ci:test:sf

**Comment by alessiorendina** [comment]
ci:test:relevant

**Comment by alessiorendina** [comment]
ci:test:sf

**Comment by alessiorendina** [comment]
ci:test:relevant

**Comment by alessiorendina** [comment]
ci:test:commerce-functional

**Comment by alessiorendina** [comment]
ci:test:sf

**Comment by alessiorendina** [comment]
ci:test:relevant

**Comment by alessiorendina** [comment]
ci:test:commerce-functional

**Comment by alessiorendina** [comment]
ci:test:commerce-functional

**Comment by alessiorendina** [comment]
ci:test:sf

**Comment by alessiorendina** [comment]
ci:test:sf

**Comment by alessiorendina** [comment]
ci:test:sf

**Comment by alessiorendina** [comment]
ci:test:sf

**Comment by alessiorendina** [comment]
ci:test:relevant

**Comment by alessiorendina** [comment]
ci:test:commerce-functional

**Comment by alessiorendina** [comment]
ci:test:sf

**Comment by alessiorendina** [comment]
ci:test:relevant

**Comment by alessiorendina** [comment]
ci:test:commerce-functional

**Comment by alessiorendina** [comment]
ci:test:sf

**Comment by alessiorendina** [comment]
ci:test:relevant

**Comment by alessiorendina** [comment]
ci:test:commerce-fucntional

**Comment by alessiorendina** [comment]
ci:test:relevant

**Comment by 22LucaGiannotta22** [comment]
relevant failures are not related to the PR. It can be force forwarded

**Comment by alessiorendina** [comment]
After a chat with @22LucaGiannotta22 , he confirmed me that we can forward the PR

**Comment by alessiorendina** [comment]
ci:forward:force

**Pull request closed without merging**